src/workerd/io/worker.c++

```diff
@@ -2575,9 +2584,10 @@ private:
     Activity activity(*this);
     Writer writer(*this);
 
-    std::unique_ptr<const v8::HeapSnapshot> snapshot(
-        js.v8Isolate->GetHeapProfiler()->TakeHeapSnapshot(&activity, nullptr,
-            exposeInternals, captureNumericValue));
+    auto profiler = js.v8Isolate->GetHeapProfiler();
+    auto snapshot = kj::Own<const v8::HeapSnapshot>(
+        profiler->TakeHeapSnapshot(&activity, nullptr, exposeInternals, captureNumericValue),
+        HeapSnapshotDeleter::INSTANCE);
     snapshot->Serialize(&writer);
```

Comment:
Why not call delete after Serialize?

Reply:
I generally prefer the consistency of the kj::Own pattern here and we're using that pattern in a number of other places.